src/UrlGenerator.php

```diff
@@ -50,19 +48,8 @@ public function __construct(
      */
     public function generate(string $name, array $parameters = []): string
     {
-        $parameters = array_map('\strval', $parameters);
+        $parameters = array_map('\strval', array_merge($this->defaults, $parameters));
```

Comment:
For performance better type cast defaults on set.

Reply:
Should not be much of a performance gain.